### PR TITLE
Fix Runtime Error for API Invocation with Security Set to 'None' When Production Endpoint is Missing

### PR DESCRIPTION
**Pull request by ashanhr**
### Purpose

> - Invoking an API resource with security set to 'None' returns a runtime error if the Production endpoint is not configured, even though the request can be routed to the Sandbox endpoint. This PR addresses and resolves this issue.
> - Add tests to validate the flows in Velocity templates when resource security is set to None.
> - Related to: https://github.com/wso2/api-manager/issues/3294

### Approach 

> - Updated the Velocity template to properly handle resources when security is set to 'None'.
> - Tested all possible scenarios across the Hybrid, Sandbox, and Production environments to ensure the fix resolves the issue without impacting other functionalities.
> -  Add two tests for the hybrid gateway environment and the sandbox gateway environment,
>    - In the hybrid gateway environment, the following scenarios are tested:
>       - Both production and sandbox endpoints are available.
>       - Only the production endpoint is available.
>       - Only the sandbox endpoint is available.
>    - In the sandbox gateway environment, the following scenarios are tested:
>       - Both production and sandbox endpoints are available.
>       - Only the sandbox endpoint is available.

### CODE REVIEW

**Review by tharikaGitHub — APPROVED** [review]

**Review by Krishanx92 — APPROVED** [review]

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/token/InvokeAPIWithVariousEndpointsAndTokensInSandboxEnvTestCase.java [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+        this.userMode = userMode;
+    }
+
+    @BeforeClass(alwaysRun = true)
+    public void setEnvironment() throws Exception {
+        super.init(userMode);
+        log.info("Test starting user mode: " + userMode);
+
+        serverConfigurationManager = new ServerConfigurationManager(gatewayContextWrk);
+        serverConfigurationManager.applyConfiguration(new File(
```
Comment (tharikaGitHub) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/token/InvokeAPIWithVariousEndpointsAndTokensInSandboxEnvTestCase.java:88]: Is this more than 120 line length?

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/token/InvokeAPIWithVariousEndpointsAndTokensInSandboxEnvTestCase.java [hunk 2]
```diff
@@ -0,0 +1,300 @@
...
+                getAMResourceLocation() + File.separator + "configFiles" + File.separator + "tokenTest" + File.separator + "apiInvokeCombinationsTest" + File.separator + "deployment.toml"));
```
Comment (tharikaGitHub) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/token/InvokeAPIWithVariousEndpointsAndTokensInSandboxEnvTestCase.java:89]: Is this more than 120 line length?

Comment (ashanhr) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/token/InvokeAPIWithVariousEndpointsAndTokensInSandboxEnvTestCase.java:89]: Fixed by https://github.com/wso2-support/product-apim/pull/2218

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/token/InvokeAPIWithVariousEndpointsAndTokensTestCase.java [hunk 3]
```diff
@@ -0,0 +1,388 @@
...
+    private String providerName;
+    private String apiId1;
+    private String apiId2;
+    private String apiId3;
+    private String applicationId;
+    ApplicationKeyDTO applicationKeyProductionDTO;
+    ApplicationKeyDTO applicationKeySandboxDTO;
+    private final String SANDBOX_RESPONSE_BODY = "HelloWSO2 from File 1_Sandbox";
+    private final String PRODUCTION_RESPONSE_BODY = "HelloWSO2 from File 1";
+    private final String SANDBOX_KEY_FOR_API_WITH_NO_SANDBOX_ENDPOINT_ERROR = "{\"code\":\"900901\",\"type\":\"Status report\",\"message\":\"Runtime Error\",\"description\":\"Sandbox key offered to the API with no sandbox endpoint\"}";
```
Comment (ashanhr) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/token/InvokeAPIWithVariousEndpointsAndTokensTestCase.java:72]: Line length is fixed by https://github.com/wso2-support/product-apim/pull/2218

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/token/InvokeAPIWithVariousEndpointsAndTokensTestCase.java [hunk 4]
```diff
@@ -0,0 +1,388 @@
...
+    private final String PRODUCTION_KEY_FOR_API_WITH_NO_PRODUCTION_ENDPOINT_ERROR = "{\"code\":\"900901\",\"type\":\"Status report\",\"message\":\"Runtime Error\",\"description\":\"Production key offered to the API with no production endpoint\"}";
```
Comment (ashanhr) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/token/InvokeAPIWithVariousEndpointsAndTokensTestCase.java:73]: Line length is fixed by https://github.com/wso2-support/product-apim/pull/2218
